src/test/java/org/rulez/demokracia/pdengine/BeatTableInitializeTest.java

```diff
@@ -0,0 +1,66 @@
+package org.rulez.demokracia.pdengine;
+
+import static org.junit.Assert.*;
+
+
+import org.junit.Before;
+import org.junit.Test;
+import org.rulez.demokracia.pdengine.annotations.TestedBehaviour;
+import org.rulez.demokracia.pdengine.annotations.TestedFeature;
+import org.rulez.demokracia.pdengine.annotations.TestedOperation;
+import org.rulez.demokracia.pdengine.dataobjects.Pair;
+import org.rulez.demokracia.pdengine.exception.ReportedException;
+import org.rulez.demokracia.pdengine.testhelpers.CreatedBeatTableForInitialization;
+
+
+@TestedFeature("Schulze method")
+@TestedOperation("compute initial beat matrix")
+@TestedBehaviour("the beat matrix contains the beats")
+public class BeatTableInitializeTest extends CreatedBeatTableForInitialization {
+	
+	@Before
+	public void setUp() throws ReportedException {
+		super.setUp();
+	}
+	
+	@Test
+	public void initialize_throws_exception_when_param_is_not_defined() {
+		assertThrows(() -> beatTable.initialize(null)
+				).assertMessageIs("Invalid castVotes");
+	}
+	
+	@Test
+	public void initialize_throws_exception_when_there_are_two_same_rank() {
```

Comment:
it is not generally an error to have the same rank for a vote.
just do not increase neither side